controller/authController.go

```diff
@@ -6,42 +6,125 @@ import (
 	"github.com/ayush/go-auth/config"
 	"log" 
 	"github.com/gin-gonic/gin"
+	"github.com/gorilla/sessions"
 )
 
+//The context "c" provides access to information about the current HTTP request.
 func SignUp(c *gin.Context) {
 	var user model.User
+
+	//The c.BindJSON -> Attempts to bind the JSON data from the request body to the user variable. 
+	//The c.BindJSON(&user) function automatically parses the JSON from the request body and populates the user struct.
 	if err := c.BindJSON(&user); err != nil {
 		log.Printf("Failed to bind JSON input for signup: %v", err)
-		c.JSON(http.StatusBadRequest, gin.H{"error": "cannot bind JSON input"})
+		c.JSON(http.StatusBadRequest, gin.H{"error": "Invalid input"})
 		return
 	}
 
 	err := user.SignUp(config.DB)
 	if err != nil {
-		log.Printf("Failed to sign up user: %v", err)
-		c.JSON(http.StatusInternalServerError, gin.H{"error": "failed to sign up"})
+		// Sends back a JSON response
+		c.JSON(500, gin.H{"error": err})
 		return
 	}
 
 	log.Printf("User %s signed up successfully", user.Username)
 	c.JSON(http.StatusOK, gin.H{"message": "signup successful"})
 }
 
+
+//Basic-Auth Login
 func Login(c *gin.Context) {
 	var user model.User
-	if err := c.BindJSON(&user); err != nil {
-		log.Printf("Failed to bind JSON input for login: %v", err)
-		c.JSON(http.StatusBadRequest, gin.H{"error": "cannot bind JSON input"})
+	if err := c.ShouldBindJSON(&user); err != nil {
+		c.JSON(http.StatusBadRequest, gin.H{"error": "Invalid input"})
 		return
 	}
 
 	err := user.Login(config.DB)
+
 	if err != nil {
-		log.Printf("Login failed for user %s: %v", user.Username, err)
-		c.JSON(http.StatusUnauthorized, gin.H{"error": "incorrect username or password"})
+		if err.Error() == "user not found"{
```

Comment:
We discussed to change this, right?

Reply:
Yes, actually main isse purra change karne vala tha but phir kuch error aa raha tha which i couldnt resolve, i will do it later, because abhi code main redundancy bhi hain